src/externs.ts

```diff
@@ -124,14 +124,9 @@ export function getGeneratedExterns(
  *
  * The mangled name is safe to use as a JavaScript identifier. It is used as a globally unique
  * prefix to scope symbols in externs file (see code below).
- *
- * @param contextFileName if given is used as the context file path to resolve the source file's
- *     name against.
  */
-export function moduleNameAsIdentifier(
-    host: AnnotatorHost, fileName: string, contextFileName?: string): string {
-  const resolved = resolveModuleName(host, contextFileName || '', fileName);
-  return host.pathToModuleName('', resolved).replace(/\./g, '$');
+export function moduleNameAsIdentifier(host: AnnotatorHost, fileName: string): string {
+  return host.pathToModuleName('', fileName).replace(/\./g, '$');
```

Comment:
reminder: we'll need to make sure the bazel rules implementation (in the host) has the same assertion here before we land this.

Reply:
Which assertion?  It appears this code works in google3 without modifications to its pathToModuleName.

Reply:
Yeah it works, but I think we should have the "all paths are absolute" assert in the g3 `pathToModuleName` as well, for safety - right?

Reply:
Ah I see what you mean, yes.  Right now I still cannot assert paths are absolute:
https://github.com/angular/tsickle/pull/901/files#diff-1310126e2af8f00760656adfc34dc400R38

Our test suite actually manages to sneak the string "goog:foo.bar" into this function, which ends up getting canonicalized into a module name like 'goog_foo.bar' (!).  I want to clean that up to but I figured this change was a good stopping point that marked the next steps clearly.